platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -196,19 +215,19 @@ public void deserialize(final SerializableDataInputStream in, final int version)
 
     public void deserialize(final SerializableDataInputStream in, final int version, final int maxTransactionCount)
             throws IOException {
+        if (version >= ClassVersion.SOFTWARE_VERSION) {
+            softwareVersion = in.readSerializable();
+        } else {
+            softwareVersion = null;
+        }
         creatorId = in.readLong();
         selfParentGen = in.readLong();
         otherParentGen = in.readLong();
         selfParentHash = in.readSerializable(false, Hash::new);
         otherParentHash = in.readSerializable(false, Hash::new);
         timeCreated = in.readInstant();
-        if (version == ClassVersion.TRANSACTION_SUBCLASSES) {
-            in.readInt(); // read serialized length
-            transactions = in.readSerializableArray(ConsensusTransactionImpl[]::new, maxTransactionCount, true);
-        } else {
-            throw new UnsupportedOperationException("Unsupported version " + version + ". Minimum supported version is "
-                    + getMinimumSupportedVersion());
-        }
+        in.readInt(); // read serialized length
```

Comment:
Perhaps now is a good time to remove this `int` from the serialization protocol, since we are ignoring its value. We can piggy-back off of the new version you added to do the necessary data migration.

We would have to add a small section to the HIP, but I don't think it would be a hard sell.

Reply:
The hip was approved.  This, I assume, would require another HIP at this point.